src/ucp/dt/datatype_iter.inl

```diff
@@ -267,10 +267,12 @@ ucp_datatype_iter_next_iov(const ucp_datatype_iter_t *dt_iter,
  */
 static UCS_F_ALWAYS_INLINE
 void ucp_datatype_iter_copy_from_next(ucp_datatype_iter_t *dt_iter,
-                                      const ucp_datatype_iter_t *next_iter)
+                                      const ucp_datatype_iter_t *next_iter,
+                                      unsigned dt_mask)
```

Comment:
when dt_mask can be IOV, but dt_iter->dt_class is not IOV?

Reply:
it's not possible, but since the mask is passed as immediate value, it allows the compiler to remove some branches